diff --git a/Src/cia402appl.c b/Srcsn/cia402appl.c
index 120922d..b29b1f2 100644
--- a/Src/cia402appl.c
+++ b/Srcsn/cia402appl.c
@@ -47,21 +47,21 @@ V4.30 : create file (state machine; handling state transition options; input fee
 ------    Includes
 ------
 -----------------------------------------------------------------------------------------*/
-#include "ecat_def.h"
-
-/*ECATCHANGE_START(V5.13) CIA402 3*/
-/*ECATCHANGE_END(V5.13) CIA402 3*/
-
 #include "applInterface.h"
-
+#include "aic_ecat_cfg.h"
 
 #include "coeappl.h"
+#include "ecat_def.h"
 
 #define _CiA402_
 #include "cia402appl.h"
 #undef _CiA402_
 
-
+#if defined(USING_ACTUAL_MOTOR) && USING_ACTUAL_MOTOR
+#include "bldc_foc.h"
+extern mcl_user_value_t motor_speed;
+extern mcl_user_value_t motor_position;
+#endif
 /*--------------------------------------------------------------------------------------
 ------
 ------    local types and defines
@@ -123,23 +123,23 @@ UINT16 CiA402_Init(void)
 
         /***set Object offset to PDO entries***/
 
-        /*csv/csp RxPDO*/
+        /*csp RxPDO*/
         for (j = 0; j < LocalAxes[AxisCnt].Objects.sRxPDOMap0.u16SubIndex0; j++)
         {
             LocalAxes[AxisCnt].Objects.sRxPDOMap0.aEntries[j] += AxisCnt* (ObjectOffset << 16);
         }
 
         /*csp RxPDO*/
-        for (j = 0; j < LocalAxes[AxisCnt].Objects.sRxPDOMap1.u16SubIndex0; j++)
-        {
-            LocalAxes[AxisCnt].Objects.sRxPDOMap1.aEntries[j] += AxisCnt* (ObjectOffset << 16);
-        }
+//        for (j = 0; j < LocalAxes[AxisCnt].Objects.sRxPDOMap1.u16SubIndex0; j++)
+//        {
+//            LocalAxes[AxisCnt].Objects.sRxPDOMap1.aEntries[j] += AxisCnt* (ObjectOffset << 16);
+//        }
 
         /*csv RxPDO*/
-        for (j = 0; j < LocalAxes[AxisCnt].Objects.sRxPDOMap2.u16SubIndex0; j++)
-        {
-            LocalAxes[AxisCnt].Objects.sRxPDOMap2.aEntries[j] += AxisCnt* (ObjectOffset << 16);
-        }
+//        for (j = 0; j < LocalAxes[AxisCnt].Objects.sRxPDOMap2.u16SubIndex0; j++)
+//        {
+//            LocalAxes[AxisCnt].Objects.sRxPDOMap2.aEntries[j] += AxisCnt* (ObjectOffset << 16);
+//        }
 
 
         /*csv/csp TxPDO*/
@@ -149,16 +149,16 @@ UINT16 CiA402_Init(void)
         }
 
         /*csp TxPDO*/
-        for (j = 0; j < LocalAxes[AxisCnt].Objects.sTxPDOMap1.u16SubIndex0; j++)
-        {
-            LocalAxes[AxisCnt].Objects.sTxPDOMap1.aEntries[j] += AxisCnt* (ObjectOffset << 16);
-        }
+//        for (j = 0; j < LocalAxes[AxisCnt].Objects.sTxPDOMap1.u16SubIndex0; j++)
+//        {
+//            LocalAxes[AxisCnt].Objects.sTxPDOMap1.aEntries[j] += AxisCnt* (ObjectOffset << 16);
+//        }
 
         /*csv TxPDO*/
-        for (j = 0; j < LocalAxes[AxisCnt].Objects.sTxPDOMap2.u16SubIndex0; j++)
-        {
-            LocalAxes[AxisCnt].Objects.sTxPDOMap2.aEntries[j] += AxisCnt* (ObjectOffset << 16);
-        }
+//        for (j = 0; j < LocalAxes[AxisCnt].Objects.sTxPDOMap2.u16SubIndex0; j++)
+//        {
+//            LocalAxes[AxisCnt].Objects.sTxPDOMap2.aEntries[j] += AxisCnt* (ObjectOffset << 16);
+//        }
 
 
 
@@ -181,19 +181,19 @@ UINT16 CiA402_Init(void)
                 pDiCEntry->pVarPtr = &LocalAxes[AxisCnt].Objects.sRxPDOMap0;
                 break;
             case 0x1601:
-                pDiCEntry->pVarPtr = &LocalAxes[AxisCnt].Objects.sRxPDOMap1;
+                //pDiCEntry->pVarPtr = &LocalAxes[AxisCnt].Objects.sRxPDOMap1;
                 break;
             case 0x1602:
-                pDiCEntry->pVarPtr = &LocalAxes[AxisCnt].Objects.sRxPDOMap2;
+                //pDiCEntry->pVarPtr = &LocalAxes[AxisCnt].Objects.sRxPDOMap2;
                 break;
             case 0x1A00:
                 pDiCEntry->pVarPtr = &LocalAxes[AxisCnt].Objects.sTxPDOMap0;
                 break;
             case 0x1A01:
-                pDiCEntry->pVarPtr = &LocalAxes[AxisCnt].Objects.sTxPDOMap1;
+                //pDiCEntry->pVarPtr = &LocalAxes[AxisCnt].Objects.sTxPDOMap1;
                 break;
             case 0x1A02:
-                pDiCEntry->pVarPtr = &LocalAxes[AxisCnt].Objects.sTxPDOMap2;
+                //pDiCEntry->pVarPtr = &LocalAxes[AxisCnt].Objects.sTxPDOMap2;
                 break;
             case 0x603F:
                 pDiCEntry->pVarPtr = &LocalAxes[AxisCnt].Objects.objErrorCode;
@@ -380,6 +380,9 @@ void CiA402_StateMachine(void)
             else if (((ControlWord6040 & CONTROLWORD_COMMAND_SWITCHON_MASK) == CONTROLWORD_COMMAND_SWITCHON) ||
                     ((ControlWord6040 & CONTROLWORD_COMMAND_SWITCHON_ENABLEOPERATION_MASK) == CONTROLWORD_COMMAND_SWITCHON_ENABLEOPERATION))
                 {
+//                    #if defined(USING_ACTUAL_MOTOR) && USING_ACTUAL_MOTOR
+//                    motor_enable();
+//                    #endif
                     pCiA402Axis->i16State = STATE_SWITCHED_ON;           // Transition 3
                 }
                 break;
@@ -388,17 +391,26 @@ void CiA402_StateMachine(void)
 
             if ((ControlWord6040 & CONTROLWORD_COMMAND_SHUTDOWN_MASK) == CONTROLWORD_COMMAND_SHUTDOWN)
             {
+                #if defined(USING_ACTUAL_MOTOR) && USING_ACTUAL_MOTOR
+                motor_disable();
+                #endif
                 pCiA402Axis->i16State = STATE_READY_TO_SWITCH_ON; // Transition 6
 
             } 
             else if (((ControlWord6040 & CONTROLWORD_COMMAND_QUICKSTOP_MASK) == CONTROLWORD_COMMAND_QUICKSTOP
                     || (ControlWord6040 & CONTROLWORD_COMMAND_DISABLEVOLTAGE_MASK) == CONTROLWORD_COMMAND_DISABLEVOLTAGE))
                 {
+                    #if defined(USING_ACTUAL_MOTOR) && USING_ACTUAL_MOTOR
+                    motor_disable();
+                    #endif
                     pCiA402Axis->i16State = STATE_SWITCH_ON_DISABLED; // Transition 10
 
                 } 
             else if ((ControlWord6040 & CONTROLWORD_COMMAND_ENABLEOPERATION_MASK) == CONTROLWORD_COMMAND_ENABLEOPERATION)
                     {
+                        #if defined(USING_ACTUAL_MOTOR) && USING_ACTUAL_MOTOR
+                        motor_enable();
+                        #endif
                         pCiA402Axis->i16State = STATE_OPERATION_ENABLED;  // Transition 4
                         //The Axis function shall be enabled and all internal set-points cleared.
                     }
@@ -428,12 +440,17 @@ void CiA402_StateMachine(void)
                             pCiA402Axis->u16PendingOptionCode = 0x605B; //STATE_TRANSITION (STATE_OPERATION_ENABLED,STATE_READY_TO_SWITCH_ON);
                             return;
                         }
-
+                        #if defined(USING_ACTUAL_MOTOR) && USING_ACTUAL_MOTOR
+                        motor_disable();
+                        #endif
                         pCiA402Axis->i16State = STATE_READY_TO_SWITCH_ON; // Transition 8
 
                     } else
                         if ((ControlWord6040 & CONTROLWORD_COMMAND_DISABLEVOLTAGE_MASK) == CONTROLWORD_COMMAND_DISABLEVOLTAGE)
                         {
+                            #if defined(USING_ACTUAL_MOTOR) && USING_ACTUAL_MOTOR
+                            motor_disable();
+                            #endif
                             pCiA402Axis->i16State = STATE_SWITCH_ON_DISABLED; // Transition 9
                         }
                         break;
@@ -450,6 +467,9 @@ void CiA402_StateMachine(void)
 
             if ((ControlWord6040 & CONTROLWORD_COMMAND_DISABLEVOLTAGE_MASK) == CONTROLWORD_COMMAND_DISABLEVOLTAGE)
             {
+                #if defined(USING_ACTUAL_MOTOR) && USING_ACTUAL_MOTOR
+                motor_disable();
+                #endif
                 pCiA402Axis->i16State = STATE_SWITCH_ON_DISABLED; // Transition 12
             }
 
@@ -472,6 +492,9 @@ void CiA402_StateMachine(void)
 
             if ((ControlWord6040 & CONTROLWORD_COMMAND_FAULTRESET_MASK) == CONTROLWORD_COMMAND_FAULTRESET)
             {
+                #if defined(USING_ACTUAL_MOTOR) && USING_ACTUAL_MOTOR
+                motor_disable();
+                #endif
                 pCiA402Axis->i16State = STATE_SWITCH_ON_DISABLED;// Transition 15
             }
             break;
@@ -563,6 +586,53 @@ void CiA402_LocalError(UINT16 ErrorCode)
  \brief    CiA402_DummyMotionControl
  \brief this functions provides an simple feedback functionality
 *////////////////////////////////////////////////////////////////////////////////////////
+#if defined(USING_ACTUAL_MOTOR) && USING_ACTUAL_MOTOR
+void CiA402_DummyMotionControl(TCiA402Axis *pCiA402Axis)
+{
+    if(pCiA402Axis->bAxisFunctionEnabled &&
+    pCiA402Axis->bLowLevelPowerApplied &&
+    pCiA402Axis->bHighLevelPowerApplied &&
+    !pCiA402Axis->bBrakeApplied)
+    {
+        if((pCiA402Axis->Objects.objSoftwarePositionLimit.i32MaxLimit> pCiA402Axis->Objects.objPositionActualValue
+            || pCiA402Axis->Objects.objPositionActualValue > pCiA402Axis->Objects.objTargetPosition) &&
+            (pCiA402Axis->Objects.objSoftwarePositionLimit.i32MinLimit < pCiA402Axis->Objects.objPositionActualValue
+            || pCiA402Axis->Objects.objPositionActualValue < pCiA402Axis->Objects.objTargetPosition))
+        {
+            pCiA402Axis->Objects.objStatusWord &= ~STATUSWORD_INTERNAL_LIMIT;
+
+            switch(pCiA402Axis->Objects.objModesOfOperationDisplay)
+            {
+            case CYCLIC_SYNC_POSITION_MODE:
+                motor_position_loop_set(pCiA402Axis->Objects.objTargetPosition);
+                break;
+            case CYCLIC_SYNC_VELOCITY_MODE:
+                    if (pCiA402Axis->i16State == STATE_OPERATION_ENABLED)
+                    {
+                        motor_speed_loop_set(pCiA402Axis->Objects.objTargetVelocity);
+                    }
+                    else
+                    {
+                        motor_speed_loop_set(0);
+                    }
+                break;
+            default:
+                break;
+            }
+        }
+        else
+        {
+            pCiA402Axis->Objects.objStatusWord |= STATUSWORD_INTERNAL_LIMIT;
+        }
+    }
+    pCiA402Axis->Objects.objVelocityActualValue= motor_get_actual_speed();
+    pCiA402Axis->Objects.objPositionActualValue = motor_get_actual_position();
+
+    /*Accept new mode of operation*/
+    pCiA402Axis->Objects.objModesOfOperationDisplay = pCiA402Axis->Objects.objModesOfOperation;
+
+}
+#else
 void CiA402_DummyMotionControl(TCiA402Axis *pCiA402Axis)
 {
     float IncFactor    = (float)0.0010922 * (float) pCiA402Axis->u32CycleTime;
@@ -625,6 +695,7 @@ void CiA402_DummyMotionControl(TCiA402Axis *pCiA402Axis)
     pCiA402Axis->Objects.objModesOfOperationDisplay = pCiA402Axis->Objects.objModesOfOperation;
 
 }
+#endif
 /////////////////////////////////////////////////////////////////////////////////////////
 /**
  \return TRUE if moving on predefined ramp is finished
@@ -1224,30 +1295,30 @@ UINT16 APPL_GenerateMapping(UINT16* pInputSize,UINT16* pOutputSize)
     {
         switch ((sRxPDOassign.aEntries[PDOAssignEntryCnt] & 0x000F))    //mask Axis type (supported modes)
         {
-            case 0:
+            case 1:
                 /*drive mode supported    csv(cyclic sync velocity) : bit 8
                                         csp (cyclic sync position) : bit 7*/
-                LocalAxes[PDOAssignEntryCnt].Objects.objSupportedDriveModes = 0x180;
-                for(u16cnt =0 ; u16cnt < LocalAxes[PDOAssignEntryCnt].Objects.sRxPDOMap0.u16SubIndex0;u16cnt++)
-                {
-                    OutputSize +=(UINT16)(LocalAxes[PDOAssignEntryCnt].Objects.sRxPDOMap0.aEntries[u16cnt] & 0xFF);
-                }
+//                LocalAxes[PDOAssignEntryCnt].Objects.objSupportedDriveModes = 0x180;
+//                for(u16cnt =0 ; u16cnt < LocalAxes[PDOAssignEntryCnt].Objects.sRxPDOMap0.u16SubIndex0;u16cnt++)
+//                {
+//                    OutputSize +=(UINT16)(LocalAxes[PDOAssignEntryCnt].Objects.sRxPDOMap0.aEntries[u16cnt] & 0xFF);
+//                }
                 break;
-            case 1:
+            case 0:
                 /*drive mode supported    csp (cyclic sync position) : bit 7*/
                 LocalAxes[PDOAssignEntryCnt].Objects.objSupportedDriveModes = 0x80;
-                for(u16cnt =0 ; u16cnt < LocalAxes[PDOAssignEntryCnt].Objects.sRxPDOMap1.u16SubIndex0;u16cnt++)
+                for(u16cnt =0 ; u16cnt < LocalAxes[PDOAssignEntryCnt].Objects.sRxPDOMap0.u16SubIndex0;u16cnt++)
                 {
-                    OutputSize +=(UINT16)(LocalAxes[PDOAssignEntryCnt].Objects.sRxPDOMap1.aEntries[u16cnt] & 0xFF);
+                    OutputSize +=(UINT16)(LocalAxes[PDOAssignEntryCnt].Objects.sRxPDOMap0.aEntries[u16cnt] & 0xFF);
                 }
                 break;
             case 2:
                     /*drive mode supported    csv(cyclic sync velocity) : bit 8*/
-                LocalAxes[PDOAssignEntryCnt].Objects.objSupportedDriveModes= 0x100;
-                for(u16cnt =0 ; u16cnt < LocalAxes[PDOAssignEntryCnt].Objects.sRxPDOMap2.u16SubIndex0;u16cnt++)
-                {
-                    OutputSize += (UINT16)(LocalAxes[PDOAssignEntryCnt].Objects.sRxPDOMap2.aEntries[u16cnt] & 0xFF);;
-                }
+//                LocalAxes[PDOAssignEntryCnt].Objects.objSupportedDriveModes= 0x100;
+//                for(u16cnt =0 ; u16cnt < LocalAxes[PDOAssignEntryCnt].Objects.sRxPDOMap2.u16SubIndex0;u16cnt++)
+//                {
+//                    OutputSize += (UINT16)(LocalAxes[PDOAssignEntryCnt].Objects.sRxPDOMap2.aEntries[u16cnt] & 0xFF);;
+//                }
                 break;
         }
     }
@@ -1261,23 +1332,23 @@ UINT16 APPL_GenerateMapping(UINT16* pInputSize,UINT16* pOutputSize)
         {
             switch ((sTxPDOassign.aEntries[PDOAssignEntryCnt] & 0x000F))    //mask Axis type (supported modes)
             {
-            case 0: /*csp/csv*/
+            case 0: /*csp*/
                 for(u16cnt =0 ; u16cnt < LocalAxes[PDOAssignEntryCnt].Objects.sTxPDOMap0.u16SubIndex0;u16cnt++)
                 {
                     InputSize +=(UINT16)(LocalAxes[PDOAssignEntryCnt].Objects.sTxPDOMap0.aEntries[u16cnt] & 0xFF);
                 }
                 break;
             case 1: /*csp*/
-                for(u16cnt =0 ; u16cnt < LocalAxes[PDOAssignEntryCnt].Objects.sTxPDOMap1.u16SubIndex0;u16cnt++)
-                {
-                    InputSize +=(UINT16)(LocalAxes[PDOAssignEntryCnt].Objects.sTxPDOMap1.aEntries[u16cnt] & 0xFF);
-                }
+//                for(u16cnt =0 ; u16cnt < LocalAxes[PDOAssignEntryCnt].Objects.sTxPDOMap1.u16SubIndex0;u16cnt++)
+//                {
+//                    InputSize +=(UINT16)(LocalAxes[PDOAssignEntryCnt].Objects.sTxPDOMap1.aEntries[u16cnt] & 0xFF);
+//                }
                 break;
             case 2: /*csv*/
-                for(u16cnt =0 ; u16cnt < LocalAxes[PDOAssignEntryCnt].Objects.sTxPDOMap2.u16SubIndex0;u16cnt++)
-                {
-                    InputSize +=(UINT16)(LocalAxes[PDOAssignEntryCnt].Objects.sTxPDOMap2.aEntries[u16cnt] & 0xFF);
-                }
+//                for(u16cnt =0 ; u16cnt < LocalAxes[PDOAssignEntryCnt].Objects.sTxPDOMap2.u16SubIndex0;u16cnt++)
+//                {
+//                    InputSize +=(UINT16)(LocalAxes[PDOAssignEntryCnt].Objects.sTxPDOMap2.aEntries[u16cnt] & 0xFF);
+//                }
                 break;
             }
         }
@@ -1309,46 +1380,46 @@ void APPL_InputMapping(UINT16* pData)
 
         switch ((sTxPDOassign.aEntries[j]& 0x000F))
         {
-        case 0:    //copy csp/csv TxPDO entries
+        case 1:    //copy csp/csv TxPDO entries
             {
-                TCiA402PDO1A00 *pInputs = (TCiA402PDO1A00 *)pTmpData;
-
-                pInputs->ObjStatusWord = SWAPWORD(LocalAxes[AxisIndex].Objects.objStatusWord);
-                pInputs->ObjPositionActualValue = SWAPDWORD(LocalAxes[AxisIndex].Objects.objPositionActualValue);
-                pInputs->ObjVelocityActualValue = SWAPDWORD(LocalAxes[AxisIndex].Objects.objVelocityActualValue);
-                pInputs->ObjModesOfOperationDisplay = SWAPWORD((LocalAxes[AxisIndex].Objects.objModesOfOperationDisplay & 0x00FF));
-
-                /*shift pointer PDO mapping object following*/
-                if(j < (sTxPDOassign.u16SubIndex0 - 1))
-                    pTmpData += SIZEOF(TCiA402PDO1A00);
+//                TCiA402PDO1A00 *pInputs = (TCiA402PDO1A00 *)pTmpData;
+//
+//                pInputs->ObjStatusWord = SWAPWORD(LocalAxes[AxisIndex].Objects.objStatusWord);
+//                pInputs->ObjPositionActualValue = SWAPDWORD(LocalAxes[AxisIndex].Objects.objPositionActualValue);
+//                pInputs->ObjVelocityActualValue = SWAPDWORD(LocalAxes[AxisIndex].Objects.objVelocityActualValue);
+//                pInputs->ObjModesOfOperationDisplay = SWAPWORD((LocalAxes[AxisIndex].Objects.objModesOfOperationDisplay & 0x00FF));
+//
+//                /*shift pointer PDO mapping object following*/
+//                if(j < (sTxPDOassign.u16SubIndex0 - 1))
+//                    pTmpData += SIZEOF(TCiA402PDO1A00);
             }
             break;
-        case 1://copy csp TxPDO entries
+        case 0://copy csp TxPDO entries
             {
-                TCiA402PDO1A01 *pInputs = (TCiA402PDO1A01 *)pTmpData;
+                TCiA402PDO1A00 *pInputs = (TCiA402PDO1A00 *)pTmpData;
 
                 pInputs->ObjStatusWord = SWAPWORD(LocalAxes[AxisIndex].Objects.objStatusWord);
                 pInputs->ObjPositionActualValue = SWAPDWORD(LocalAxes[AxisIndex].Objects.objPositionActualValue);
-
+                pInputs->ObjModesOfOperationDisplay = SWAPDWORD(LocalAxes[AxisIndex].Objects.objModesOfOperationDisplay);
                 /*shift pointer PDO mapping object following*/
                 if (j < (sTxPDOassign.u16SubIndex0 - 1))
                 {
-                    pTmpData += SIZEOF(TCiA402PDO1A01);
+                    pTmpData += SIZEOF(TCiA402PDO1A00);
                 }
             }
             break;
         case 2://copy csv TxPDO entries
             {
-                TCiA402PDO1A02 *pInputs = (TCiA402PDO1A02 *)pTmpData;
-
-                pInputs->ObjStatusWord = SWAPWORD(LocalAxes[AxisIndex].Objects.objStatusWord);
-                pInputs->ObjPositionActualValue = SWAPDWORD(LocalAxes[AxisIndex].Objects.objPositionActualValue);
-            
-                /*shift pointer PDO mapping object following*/
-                if (j < (sTxPDOassign.u16SubIndex0 - 1))
-                {
-                    pTmpData += SIZEOF(TCiA402PDO1A02);
-                }
+//                TCiA402PDO1A02 *pInputs = (TCiA402PDO1A02 *)pTmpData;
+//
+//                pInputs->ObjStatusWord = SWAPWORD(LocalAxes[AxisIndex].Objects.objStatusWord);
+//                pInputs->ObjPositionActualValue = SWAPDWORD(LocalAxes[AxisIndex].Objects.objPositionActualValue);
+//
+//                /*shift pointer PDO mapping object following*/
+//                if (j < (sTxPDOassign.u16SubIndex0 - 1))
+//                {
+//                    pTmpData += SIZEOF(TCiA402PDO1A02);
+//                }
             }
             break;
         }    //switch TXPDO entry
@@ -1375,48 +1446,48 @@ void APPL_OutputMapping(UINT16* pData)
 
         switch ((sRxPDOassign.aEntries[j] & 0x000F))
         {
-        case 0:    //csp/csv RxPDO    entries
+        case 1:    //csp/csv RxPDO    entries
             {
-            TCiA402PDO1600 *pOutputs = (TCiA402PDO1600 *)pTmpData;
-
-            LocalAxes[AxisIndex].Objects.objControlWord = SWAPWORD(pOutputs->ObjControlWord);
-            LocalAxes[AxisIndex].Objects.objTargetPosition = SWAPDWORD(pOutputs->ObjTargetPosition);
-            LocalAxes[AxisIndex].Objects.objTargetVelocity    = SWAPDWORD(pOutputs->ObjTargetVelocity);
-            LocalAxes[AxisIndex].Objects.objModesOfOperation = SWAPWORD((pOutputs->ObjModesOfOperation & 0x00FF));
-
-            /*shift pointer PDO mapping object following*/
-            if (j < (sRxPDOassign.u16SubIndex0 - 1))
-            {
-                pTmpData += SIZEOF(TCiA402PDO1600);
-            }
+//            TCiA402PDO1600 *pOutputs = (TCiA402PDO1600 *)pTmpData;
+//
+//            LocalAxes[AxisIndex].Objects.objControlWord = SWAPWORD(pOutputs->ObjControlWord);
+//            LocalAxes[AxisIndex].Objects.objTargetPosition = SWAPDWORD(pOutputs->ObjTargetPosition);
+//            LocalAxes[AxisIndex].Objects.objTargetVelocity    = SWAPDWORD(pOutputs->ObjTargetVelocity);
+//            LocalAxes[AxisIndex].Objects.objModesOfOperation = SWAPWORD((pOutputs->ObjModesOfOperation & 0x00FF));
+//
+//            /*shift pointer PDO mapping object following*/
+//            if (j < (sRxPDOassign.u16SubIndex0 - 1))
+//            {
+//                pTmpData += SIZEOF(TCiA402PDO1600);
+//            }
             }
             break;
-        case 1:    //csp RxPDO    entries
+        case 0:    //csp RxPDO    entries
             {
-            TCiA402PDO1601 *pOutputs = (TCiA402PDO1601 *)pTmpData;
+            TCiA402PDO1600 *pOutputs = (TCiA402PDO1600 *)pTmpData;
 
             LocalAxes[AxisIndex].Objects.objControlWord = SWAPWORD(pOutputs->ObjControlWord);
             LocalAxes[AxisIndex].Objects.objTargetPosition = SWAPDWORD(pOutputs->ObjTargetPosition);
-
+            LocalAxes[AxisIndex].Objects.objModesOfOperation = SWAPDWORD(pOutputs->ObjModesOfOperation);//CYCLIC_SYNC_POSITION_MODE;
             /*shift pointer PDO mapping object following*/
             if (j < (sRxPDOassign.u16SubIndex0 - 1))
             {
-                pTmpData += SIZEOF(TCiA402PDO1601);
+                pTmpData += SIZEOF(TCiA402PDO1600);
             }
             }
             break;
         case 2:    //csv RxPDO    entries
             {
-            TCiA402PDO1602 *pOutputs = (TCiA402PDO1602 *)pTmpData;
-
-            LocalAxes[AxisIndex].Objects.objControlWord = SWAPWORD(pOutputs->ObjControlWord);
-            LocalAxes[AxisIndex].Objects.objTargetVelocity    = SWAPDWORD(pOutputs->ObjTargetVelocity);
-
-            /*shift pointer PDO mapping object following*/
-            if (j < (sRxPDOassign.u16SubIndex0 - 1))
-            {
-                pTmpData += SIZEOF(TCiA402PDO1602);
-            }
+//            TCiA402PDO1602 *pOutputs = (TCiA402PDO1602 *)pTmpData;
+//
+//            LocalAxes[AxisIndex].Objects.objControlWord = SWAPWORD(pOutputs->ObjControlWord);
+//            LocalAxes[AxisIndex].Objects.objTargetVelocity    = SWAPDWORD(pOutputs->ObjTargetVelocity);
+//
+//            /*shift pointer PDO mapping object following*/
+//            if (j < (sRxPDOassign.u16SubIndex0 - 1))
+//            {
+//                pTmpData += SIZEOF(TCiA402PDO1602);
+//            }
             }
             break;
         }
@@ -1441,6 +1512,86 @@ void APPL_Application(void)
 }
 
 
+#if defined(USING_ACTUAL_MOTOR) && USING_ACTUAL_MOTOR
+UINT8 ObjRead0x6064( UINT16 Index, UINT8 Subindex, UINT32 Size, UINT16 MBXMEM * pData, UINT8 bCompleteAccess)
+{
+    if(bCompleteAccess)
+    {
+        return ABORTIDX_UNSUPPORTED_ACCESS;
+    }
+
+    if(Subindex == 0)
+    {
+        *(INT32 *)pData = motor_get_actual_position();
+    } else {
+        return ABORTIDX_SUBINDEX_NOT_EXISTING;
+    }
+ return 0;
+}
+
+UINT8 ObjRead0x606c( UINT16 Index, UINT8 Subindex, UINT32 Size, UINT16 MBXMEM * pData, UINT8 bCompleteAccess)
+{
+    if(bCompleteAccess)
+    {
+        return ABORTIDX_UNSUPPORTED_ACCESS;
+    }
+
+    if(Subindex == 0)
+    {
+            *(UINT32 *)pData = motor_get_actual_speed();
+    } else {
+        return ABORTIDX_SUBINDEX_NOT_EXISTING;
+    }
+ return 0;
+}
+
+UINT8 ObjRead0x6060( UINT16 Index, UINT8 Subindex, UINT32 Size, UINT16 MBXMEM * pData, UINT8 bCompleteAccess)
+{
+    if(bCompleteAccess)
+    {
+        return ABORTIDX_UNSUPPORTED_ACCESS;
+    }
+
+    if(Subindex == 0)
+    {
+
+        if (motor_speed.enable)
+            *pData = CYCLIC_SYNC_VELOCITY_MODE;
+        else if (motor_position.enable)
+            *pData = CYCLIC_SYNC_POSITION_MODE;
+        else
+            *pData = NO_MODE;
+
+    } else {
+        return ABORTIDX_SUBINDEX_NOT_EXISTING;
+    }
+ return 0;
+}
+
+UINT8 ObjWrite0x6060( UINT16 Index, UINT8 Subindex, UINT32 Size, UINT16 MBXMEM * pData, UINT8 bCompleteAccess)
+{
+   if ( bCompleteAccess )
+   {
+      /* Complete Access is not supported for object 0x1010 */
+      return ABORTIDX_UNSUPPORTED_ACCESS;
+   }
+
+   if ( Subindex == 0 )
+   {
+      if (*pData == CYCLIC_SYNC_POSITION_MODE) {
+          motor_postion_loop_init();
+          LocalAxes[0].Objects.objModesOfOperation = CYCLIC_SYNC_POSITION_MODE;
+      } else if (*pData == CYCLIC_SYNC_VELOCITY_MODE) {
+          motor_speed_loop_init();
+          LocalAxes[0].Objects.objModesOfOperation = CYCLIC_SYNC_VELOCITY_MODE;
+      } else
+          return ABORTIDX_PARAM_IS_INCOMPATIBLE;
+   } else {
+      return ABORTIDX_SUBINDEX_NOT_EXISTING;
+   }
+ return 0;
+}
+#endif
 
 /** @} */
 
diff --git a/Src/cia402appl.h b/Srcsn/cia402appl.h
index ccb754e..e413529 100644
--- a/Src/cia402appl.h
+++ b/Srcsn/cia402appl.h
@@ -324,7 +324,7 @@ V5.01 : Start file change log
 //9 to 32767        RESERVED
 
 
-#define MAX_AXES    2 /**< \brief Number of supported axes*/
+#define MAX_AXES    1 /**< \brief Number of supported axes*/
 
 
 
@@ -338,62 +338,26 @@ V5.01 : Start file change log
 
 /**
  * \addtogroup PdoMappingObjects PDO Mapping Objects
- * csp/csv mode RxPDO mapping : 0x1600<br>
- * csp     mode RxPDO mapping : 0x1601<br>
- * csv     mode RxPDO mapping : 0x1602<br>
- * csp/csv mode TxPDO mapping : 0x1A00<br>
- * csp     mode TxPDO mapping : 0x1A01<br>
- * csv     mode TxPDO mapping : 0x1A02
+ * csp     mode RxPDO mapping : 0x1600<br>
+ * csp     mode TxPDO mapping : 0x1A00<br>
  * @{
  */
-/** \brief 0x1600 (csp/csv RxPDO) data structure*/
-typedef struct OBJ_STRUCT_PACKED_START {
-   UINT16   u16SubIndex0; /**< \brief SubIndex 0*/
-   UINT32   aEntries[5]; /**< \brief Entry buffer*/
-} OBJ_STRUCT_PACKED_END
-TOBJ1600;
-
-
-/** \brief 0x1601 (csp RxPDO) data structure*/
-typedef struct OBJ_STRUCT_PACKED_START {
-   UINT16   u16SubIndex0; /**< \brief SubIndex 0*/
-   UINT32   aEntries[3]; /**< \brief Entry buffer*/
-} OBJ_STRUCT_PACKED_END
-TOBJ1601;
 
-
-/** \brief 0x1602 (csv RxPDO) data structure*/
+/** \brief 0x1600 (csp RxPDO) data structure*/
 typedef struct OBJ_STRUCT_PACKED_START {
    UINT16   u16SubIndex0; /**< \brief SubIndex 0*/
-   UINT32   aEntries[3]; /**< \brief Entry buffer*/
+   UINT32   aEntries[8]; /**< \brief Entry buffer*/
 } OBJ_STRUCT_PACKED_END
-TOBJ1602;
-
+TOBJ1600;
 
-/** \brief 0x1A00 (csp/csv TxPDO) data structure*/
+/** \brief 0x1A00 (csp TxPDO) data structure*/
 typedef struct OBJ_STRUCT_PACKED_START {
    UINT16   u16SubIndex0; /**< \brief SubIndex 0*/
-   UINT32   aEntries[5]; /**< \brief Entry buffer*/
+   UINT32   aEntries[8]; /**< \brief Entry buffer*/
 } OBJ_STRUCT_PACKED_END
 TOBJ1A00;
 
 
-/** \brief 0x1A01 (csp TxPDO) data structure*/
-typedef struct OBJ_STRUCT_PACKED_START {
-   UINT16   u16SubIndex0; /**< \brief SubIndex 0*/
-   UINT32   aEntries[3]; /**< \brief Entry buffer*/
-} OBJ_STRUCT_PACKED_END
-TOBJ1A01;
-
-
-/** \brief 0x1A02 (csv TxPDO) data structure*/
-typedef struct OBJ_STRUCT_PACKED_START {
-   UINT16   u16SubIndex0; /**< \brief SubIndex 0*/
-   UINT32   aEntries[3]; /**< \brief Entry buffer*/
-} OBJ_STRUCT_PACKED_END
-TOBJ1A02;
-/** @}*/
-
 
 /**
  * \addtogroup SmAssignObjects SyncManager Assignment Objects
@@ -502,69 +466,27 @@ TOBJF050;
  * \addtogroup PDO Process Data Objects
  * @{
  */
-/** \brief Data structure to handle the process data transmitted via 0x1A00 (csp/csv TxPDO)*/
+
+/** \brief Data structure to handle the process data transmitted via 0x1A00 (csp TxPDO)*/
 typedef struct STRUCT_PACKED_START
 {
     UINT16 ObjStatusWord; /**< \brief Status word (0x6041)*/
     INT32 ObjPositionActualValue; /**< \brief Actual position (0x6064)*/
-    INT32 ObjVelocityActualValue; /**< \brief Actual velocity (0x606C)*/
     INT16 ObjModesOfOperationDisplay; /**< \brief Current mode of operation (0x6061)*/
 }STRUCT_PACKED_END
 TCiA402PDO1A00;
 
 
-/** \brief Data structure to handle the process data transmitted via 0x1A01 (csp TxPDO)*/
-typedef struct STRUCT_PACKED_START
-{
-    UINT16 ObjStatusWord; /**< \brief Status word (0x6041)*/
-    INT32 ObjPositionActualValue; /**< \brief Actual position (0x6064)*/
-    UINT16 Padding16Bit; /**< \brief 16bit padding*/
-}STRUCT_PACKED_END
-TCiA402PDO1A01;
-
-
-/** \brief Data structure to handle the process data transmitted via 0x1A02 (csv TxPDO)*/
-typedef struct STRUCT_PACKED_START
-{
-    UINT16 ObjStatusWord; /**< \brief Status word (0x6041)*/
-    INT32 ObjPositionActualValue; /**< \brief Actual position (0x6064)*/
-    UINT16 Padding16Bit; /**< \brief 16bit padding*/
-}STRUCT_PACKED_END
-TCiA402PDO1A02;
-
-
-/** \brief Data structure to handle the process data transmitted via 0x1600 (csp/csv RxPDO)*/
+/** \brief Data structure to handle the process data transmitted via 0x1600 (csp RxPDO)*/
 typedef struct STRUCT_PACKED_START
 {
     UINT16 ObjControlWord; /**< \brief Control word (0x6040)*/
     INT32 ObjTargetPosition; /**< \brief Target position (0x607A)*/
-    INT32 ObjTargetVelocity; /**< \brief Target velocity (0x60FF)*/
     INT16 ObjModesOfOperation; /**< \brief Mode of operation (0x6060)*/
 }STRUCT_PACKED_END
 TCiA402PDO1600;
 
 
-/** \brief Data structure to handle the process data transmitted via 0x1601 (csp RxPDO)*/
-typedef struct STRUCT_PACKED_START
-{
-    UINT16 ObjControlWord; /**< \brief Control word (0x6040)*/
-    INT32 ObjTargetPosition; /**< \brief Target position (0x607A)*/
-    UINT16 Padding16Bit; /**< \brief 16bit padding*/
-}STRUCT_PACKED_END
-TCiA402PDO1601;
-
-
-/** \brief Data structure to handle the process data transmitted via 0x1602 (csv RxPDO)*/
-typedef struct STRUCT_PACKED_START
-{
-    UINT16 ObjControlWord; /**< \brief Control word (0x6040)*/
-    INT32 ObjTargetVelocity; /**< \brief Target velocity (0x60FF)*/
-    UINT16 Padding16Bit; /**< \brief 16bit padding*/
-}STRUCT_PACKED_END
-TCiA402PDO1602;
-/** @}*/
-
-
 
 /**
  * \addtogroup CiA402Objects CiA402 Axis Objects
@@ -573,12 +495,8 @@ TCiA402PDO1602;
 /** \brief Data structure to handle the axis specific object data*/
 typedef struct OBJ_STRUCT_PACKED_START
 {
-    TOBJ1600 sRxPDOMap0; /**< \brief csv/csp RxPDO (0x1600)*/
-    TOBJ1601 sRxPDOMap1; /**< \brief csp RxPDO (0x1601)*/
-    TOBJ1602 sRxPDOMap2; /**< \brief csv RxPDO (0x1602)*/
-    TOBJ1A00 sTxPDOMap0; /**< \brief csv/csp TxPDO (0x1A00)*/
-    TOBJ1A01 sTxPDOMap1; /**< \brief csp TxPDO (0x1A01)*/
-    TOBJ1A02 sTxPDOMap2; /**< \brief csv TxPDO (0x1A02)*/
+    TOBJ1600 sRxPDOMap0; /**< \brief csp RxPDO (0x1600)*/
+    TOBJ1A00 sTxPDOMap0; /**< \brief csp TxPDO (0x1A00)*/
 
     UINT16 objErrorCode; /**< \brief Error Code (0x603F)*/
     UINT16 objControlWord; /**< \brief Control Word (0x6040)*/
@@ -677,119 +595,59 @@ OBJCONST UCHAR OBJMEM aName0x1C13[] = "TxPDO assign";
 * \addtogroup PdoMappingObjects PDO Mapping Objects
 * @{
 */
-/**
- * \brief Object 0x1600 (csp/csv RxPDO) entry descriptions
- */
-OBJCONST TSDOINFOENTRYDESC	OBJMEM asEntryDesc0x1600[] = {
-   {DEFTYPE_UNSIGNED8, 0x8, ACCESS_READ }, /* Subindex 000 */
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}, /* SubIndex 001*/
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}, /* SubIndex 002*/
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}, /* SubIndex 003*/
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}, /* SubIndex 004*/
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}}; /* SubIndex 005*/
 
 
 /**
- * \brief Object 0x1600 (csp/csv RxPDO) object and entry names
- *
- * In this example no specific entry name is defined ("SubIndex xxx" is used)
- */   
-OBJCONST UCHAR OBJMEM aName0x1600[] = "csp/csv RxPDO\000\377";
-
-
-/**
- * \brief Object 0x1601 (csp RxPDO) entry descriptions
+ * \brief Object 0x1600 (csp RxPDO) entry descriptions
  */
-OBJCONST TSDOINFOENTRYDESC	OBJMEM asEntryDesc0x1601[] = {
-   {DEFTYPE_UNSIGNED8, 0x8, ACCESS_READ }, /* Subindex 000 */
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}, /* SubIndex 001*/
-   { DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ }, /* SubIndex 002*/
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}}; /* SubIndex 003*/
+OBJCONST TSDOINFOENTRYDESC	OBJMEM asEntryDesc0x1600[] = {
+   {DEFTYPE_UNSIGNED8, 0x8, ACCESS_READ | ACCESS_WRITE_PREOP}, /* Subindex 000 */
+   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ | ACCESS_WRITE_PREOP}, /* SubIndex 001*/
+   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ | ACCESS_WRITE_PREOP}, /* SubIndex 002*/
+   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ | ACCESS_WRITE_PREOP}, /* SubIndex 003*/
+   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ | ACCESS_WRITE_PREOP}, /* SubIndex 004*/
+   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ | ACCESS_WRITE_PREOP}, /* SubIndex 005*/
+   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ | ACCESS_WRITE_PREOP}, /* SubIndex 006*/
+   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ | ACCESS_WRITE_PREOP}, /* SubIndex 007*/
+   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ | ACCESS_WRITE_PREOP}, /* SubIndex 008*/
+
+};
 
 
 /**
- * \brief Object 0x1601 (csp RxPDO) object and entry names
+ * \brief Object 0x1600 (csp RxPDO) object and entry names
  *
  * In this example no specific entry name is defined ("SubIndex xxx" is used)
  */   
-OBJCONST UCHAR OBJMEM aName0x1601[] = "csp RxPDO\000\377";
+OBJCONST UCHAR OBJMEM aName0x1600[] = "csp RxPDO\000\377";
 
 
-/**
- * \brief Object 0x1602 (csv RxPDO) entry descriptions
- */
-OBJCONST TSDOINFOENTRYDESC	OBJMEM asEntryDesc0x1602[] = {
-   {DEFTYPE_UNSIGNED8, 0x8, ACCESS_READ }, /* Subindex 000 */
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}, /* SubIndex 001*/
-   { DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ }, /* SubIndex 002*/
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}}; /* SubIndex 003*/
-
-
-/**
- * \brief Object 0x1602 (csv RxPDO) object and entry names
- *
- * In this example no specific entry name is defined ("SubIndex xxx" is used)
- */      
-OBJCONST UCHAR OBJMEM aName0x1602[] = "csv RxPDO\000\377";
 
 
 
 /**
- * \brief Object 0x1A00 (csp/csv TxPDO) entry descriptions
+ * \brief Object 0x1A00 (csp RxPDO) entry descriptions
  */
 OBJCONST TSDOINFOENTRYDESC	OBJMEM asEntryDesc0x1A00[] = {
-   {DEFTYPE_UNSIGNED8, 0x8, ACCESS_READ }, /* Subindex 000 */
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}, /* SubIndex 001*/
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}, /* SubIndex 002*/
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}, /* SubIndex 003*/
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}, /* SubIndex 004*/
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}}; /* SubIndex 005 */
-
-   
-/**
- * \brief Object 0x1A00 (csp/csv TxPDO) object and entry names
- *
- * In this example no specific entry name is defined ("SubIndex xxx" is used)
- */      
-OBJCONST UCHAR OBJMEM aName0x1A00[] = "csp/csv TxPDO\000\377";
-
-
-/**
- * \brief Object 0x1A01 (csp RxPDO) entry descriptions
- */
-OBJCONST TSDOINFOENTRYDESC	OBJMEM asEntryDesc0x1A01[] = {
-   {DEFTYPE_UNSIGNED8, 0x8, ACCESS_READ }, /* Subindex 000 */
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}, /* SubIndex 001*/
-   { DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ }, /* SubIndex 002*/
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}}; /* SubIndex 003 */
-
-
-/**
- * \brief Object 0x1A01 (csp TxPDO) object and entry names
- *
- * In this example no specific entry name is defined ("SubIndex xxx" is used)
- */      
-OBJCONST UCHAR OBJMEM aName0x1A01[] = "csp TxPDO\000\377";
+   {DEFTYPE_UNSIGNED8, 0x8, ACCESS_READ | ACCESS_WRITE_PREOP}, /* Subindex 000 */
+   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ | ACCESS_WRITE_PREOP}, /* SubIndex 001*/
+   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ | ACCESS_WRITE_PREOP}, /* SubIndex 002*/
+   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ | ACCESS_WRITE_PREOP}, /* SubIndex 003 */
+   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ | ACCESS_WRITE_PREOP}, /* SubIndex 004*/
+   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ | ACCESS_WRITE_PREOP}, /* SubIndex 005*/
+   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ | ACCESS_WRITE_PREOP}, /* SubIndex 006 */
+   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ | ACCESS_WRITE_PREOP}, /* SubIndex 007*/
+   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ | ACCESS_WRITE_PREOP}, /* SubIndex 008 */
 
+};
 
 
 /**
- * \brief Object 0x1A02 (csv RxPDO) entry descriptions
- */
-OBJCONST TSDOINFOENTRYDESC	OBJMEM asEntryDesc0x1A02[] = {
-   {DEFTYPE_UNSIGNED8, 0x8, ACCESS_READ }, /* Subindex 000 */
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}, /* SubIndex 001*/
-   { DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ }, /* SubIndex 002*/
-   {DEFTYPE_UNSIGNED32, 0x20, ACCESS_READ}}; /* SubIndex 003*/
-
-   
-/**
- * \brief Object 0x1A02 (csv TxPDO) object and entry names
+ * \brief Object 0x1A00 (csp TxPDO) object and entry names
  *
  * In this example no specific entry name is defined ("SubIndex xxx" is used)
  */      
-OBJCONST UCHAR OBJMEM aName0x1A02[] = "csv TxPDO\000\377";
-/** @}*/
+OBJCONST UCHAR OBJMEM aName0x1A00[] = "csp TxPDO\000\377";
 
 
 
@@ -1026,12 +884,12 @@ OBJCONST UCHAR OBJMEM aName0xF050[] = "Module detected list";
  * \brief 0x1C12 (SyncManager 2 assignment) variable to handle object data
  * 
  * SubIndex 0 : 1<br>
- * SubIndex 1 : 0x1602
+ * SubIndex 1 : 0x1600
  * SubIndex 2 : 0
  */
 PROTO TOBJ1C12 sRxPDOassign
 #ifdef _CiA402_
-= {1,{0x1602,0x0}}
+= {1,{0x1600,0x0}}
 #endif
 ;
 
@@ -1045,7 +903,7 @@ PROTO TOBJ1C12 sRxPDOassign
  */
 PROTO TOBJ1C13 sTxPDOassign
 #ifdef _CiA402_
-= {1,{0x1A02,0x0}}
+= {1,{0x1A00,0x0}}
 #endif
 ;
 /** @}*/
@@ -1078,7 +936,7 @@ PROTO TOBJF000 sModulardeviceprofile
  */
 PROTO TOBJF010 sModuleProfileInfo
 #ifdef _CiA402_
-= {1,{2,0}}
+= {1,{2}}
 #endif
 ;
 
@@ -1091,7 +949,7 @@ PROTO TOBJF010 sModuleProfileInfo
  */
 PROTO TOBJF030 sConfiguredModuleIdentList
 #ifdef _CiA402_
-= {1,{0x319800,0}}
+= {1,{0x219800}}
 #endif
 ;
 
@@ -1107,7 +965,7 @@ PROTO UINT8 Write0xF030( UINT16 index, UINT8 subindex, UINT32 dataSize, UINT16 M
  */
 PROTO TOBJF050 sDetectedModuleIdentList
 #ifdef _CiA402_
-= {1,{0x319800,0}}
+= {1,{0x219800}}
 #endif
 ;
 /** @}*/
@@ -1124,12 +982,8 @@ PROTO TOBJF050 sDetectedModuleIdentList
 PROTO CiA402Objects DefCiA402ObjectValues
 #ifdef _CiA402_
 = {
-{5, {0x60400010,0x607A0020,0x60FF0020,0x60600008,0x00000008}},    /* TOBJ1600*/
-{3, {0x60400010,0x607A0020,0x00000010 }}, /*TOBJ1601*/
-{3, {0x60400010,0x60FF0020,0x00000010 }}, /*TOBJ1602*/
-{5, {0x60410010,0x60640020,0x606C0020,0x60610008,0x00000008}}, /*TOBJ1A00*/
-{3, {0x60410010,0x60640020,0x00000010 }}, /*TOBJ1A01*/
-{3, {0x60410010,0x60640020,0x00000010 }},/*TOBJ1A02*/
+{8, {0x60400010,0x607A0020,0x60600008,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000}}, /*TOBJ1600*/
+{8, {0x60410010,0x60640020,0x60610008,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000}}, /*TOBJ1A00*/
 0x0,/*(UINT16) ErrorCode 0x1001*/
 0x0,/*(UINT16) ControlWord 0x6040*/
 0x0,/*(UINT16) StatusWord 0x6041*/
@@ -1163,14 +1017,6 @@ PROTO TOBJECT    OBJMEM ApplicationObjDic[]
    {NULL,NULL, 0x1C12, {DEFTYPE_UNSIGNED16, MAX_AXES | (OBJCODE_ARR << 8)}, asPDOAssignEntryDesc, aName0x1C12, &sRxPDOassign, NULL, NULL, 0x0000 },
    /* Object 0x1C13 */
    {NULL,NULL, 0x1C13, {DEFTYPE_UNSIGNED16, MAX_AXES | (OBJCODE_ARR << 8)}, asPDOAssignEntryDesc, aName0x1C13, &sTxPDOassign, NULL, NULL, 0x0000 },
-    /* Object 0xF000 */
-   {NULL,NULL,   0xF000, {DEFTYPE_RECORD, 2 | (OBJCODE_REC << 8)}, asEntryDesc0xF000, aName0xF000, &sModulardeviceprofile, NULL, NULL, 0x0000 },
-    /* Object 0xF010 */
-   {NULL,NULL,   0xF010, {DEFTYPE_UNSIGNED32, MAX_AXES | (OBJCODE_ARR << 8)}, asEntryDesc0xF010, aName0xF010, &sModuleProfileInfo, NULL, NULL, 0x0000 },
-    /* Object 0xF030 */
-   {NULL,NULL,   0xF030, {DEFTYPE_UNSIGNED32, MAX_AXES | (OBJCODE_ARR << 8)}, asEntryDesc0xF030, aName0xF030, &sConfiguredModuleIdentList, NULL, Write0xF030, 0x0000 },
-   /* Object 0xF050 */
-   {NULL,NULL, 0xF050, {DEFTYPE_UNSIGNED32, MAX_AXES | (OBJCODE_ARR << 8)}, asEntryDesc0xF050, aName0xF050, &sDetectedModuleIdentList, NULL, NULL, 0x0000 },
    {NULL,NULL, 0xFFFF, {0, 0}, NULL, NULL, NULL, NULL, NULL, 0x000}}
 #endif
 ;
@@ -1183,18 +1029,10 @@ PROTO TOBJECT    OBJMEM ApplicationObjDic[]
 PROTO TOBJECT    OBJMEM DefCiA402AxisObjDic[]
 #ifdef _CiA402_
 = {
-    /* Object 0x1600 */
-    {NULL,NULL, 0x1600, {DEFTYPE_PDOMAPPING, 5 | (OBJCODE_REC << 8)}, asEntryDesc0x1600, aName0x1600, NULL, NULL, NULL, 0x0000 },
-    /* Object 0x1601 */
-   {NULL,NULL, 0x1601, {DEFTYPE_PDOMAPPING, 3 | (OBJCODE_REC << 8)}, asEntryDesc0x1601, aName0x1601,NULL, NULL, NULL, 0x0000 },
-    /* Object 0x1602 */
-   {NULL,NULL, 0x1602, {DEFTYPE_PDOMAPPING, 3 | (OBJCODE_REC << 8)}, asEntryDesc0x1602, aName0x1602, NULL, NULL, NULL, 0x0000 },
+   /* Object 0x1600 */
+   {NULL,NULL, 0x1600, {DEFTYPE_PDOMAPPING, 8 | (OBJCODE_REC << 8)}, asEntryDesc0x1600, aName0x1600, NULL, NULL, NULL, 0x0000 },
    /* Object 0x1A00 */
-   {NULL,NULL, 0x1A00, {DEFTYPE_PDOMAPPING, 5 | (OBJCODE_REC << 8)}, asEntryDesc0x1A00, aName0x1A00, NULL, NULL, NULL, 0x0000 },	
-   /* Object 0x1A01 */
-   {NULL,NULL, 0x1A01, {DEFTYPE_PDOMAPPING, 3 | (OBJCODE_REC << 8)}, asEntryDesc0x1A01, aName0x1A01, NULL, NULL, NULL, 0x0000 },	
-   /* Object 0x1A02 */
-   {NULL,NULL, 0x1A02, {DEFTYPE_PDOMAPPING, 3 | (OBJCODE_REC << 8)}, asEntryDesc0x1A02, aName0x1A02, NULL, NULL, NULL, 0x0000 },	
+   {NULL,NULL, 0x1A00, {DEFTYPE_PDOMAPPING, 8 | (OBJCODE_REC << 8)}, asEntryDesc0x1A00, aName0x1A00, NULL, NULL, NULL, 0x0000 },
    /* Object 0x603F */
    {NULL,NULL, 0x603F, {DEFTYPE_UNSIGNED16 , 0 | (OBJCODE_VAR << 8)}, &sEntryDesc0x603F, aName0x603F, NULL, NULL, NULL, 0x0000 },
    /* Object 0x6040 */
diff --git a/Src/ecatslv.c b/Srcsn/ecatslv.c
index 0e50535..f5a16c7 100644
--- a/Src/ecatslv.c
+++ b/Srcsn/ecatslv.c
@@ -267,7 +267,7 @@ void ResetALEventMask(UINT16 intMask)
     UINT32 u32Mask = 0;
     HW_EscReadDWord(u32Mask, ESC_AL_EVENTMASK_OFFSET);
     u32Mask &= (UINT32)intMask;
-
+    u32Mask &= ~(SYNC0_EVENT | SYNC1_EVENT); /* mask Sync0\Sync1 for PDI IRQ, because there are independent Sync0\Sync1 IRQ */
 
 
     DISABLE_ESC_INT();
@@ -287,7 +287,7 @@ void SetALEventMask(UINT16 intMask)
     UINT32 u32Mask = 0;
     HW_EscReadDWord(u32Mask, ESC_AL_EVENTMASK_OFFSET);
     u32Mask |= (UINT32)intMask;
-
+    u32Mask &= ~(SYNC0_EVENT | SYNC1_EVENT); /* mask Sync0\Sync1 for PDI IRQ, because there are independent Sync0\Sync1 IRQ */
 
     DISABLE_ESC_INT();
 
